stl/inc/algorithm

```diff
@@ -7405,6 +7609,339 @@ void inplace_merge(_ExPo&&, _BidIt _First, _BidIt _Mid, _BidIt _Last) noexcept /
+                _Pass_fn(_Pred), _Pass_fn(_Proj));
+            return _First;
+        }
+
+    private:
+        template <class _It, class _Pr, class _Pj>
+        static void _Inplace_merge_common(_It _First, _It _Mid, _It _Last, _Pr _Pred, _Pj _Proj) {
+            _STL_INTERNAL_STATIC_ASSERT(bidirectional_iterator<_It>);
+            _STL_INTERNAL_STATIC_ASSERT(sortable<_It, _Pr, _Pj>);
+
+            if (_First == _Mid || _Mid == _Last) {
+                return;
+            }
+#if _ITERATOR_DEBUG_LEVEL == 2
+            _STL_VERIFY(_RANGES _Is_sorted_until_unchecked(_First, _Mid, _Pred, _Proj) == _Mid,
+                "ranges::inplace_merge requires the range [first, middle) to be sorted");
+            _STL_VERIFY(_RANGES _Is_sorted_until_unchecked(_Mid, _Last, _Pred, _Proj) == _Last,
+                "ranges::inplace_merge requires the range [middle, last)] to be sorted");
+#endif //_ITERATOR_DEBUG_LEVEL == 2
+
+            // Find first element in [_First, _Mid) that is not less than *_Mid
+            while (!_STD invoke(_Pred, _STD invoke(_Proj, *_Mid), _STD invoke(_Proj, *_First))) {
+                if (++_First == _Mid) {
+                    return;
+                }
+            }
+
+            // Fast early return if there is only one element to be moved
+            if (_Mid == --_Last) {
+                // rotate only element remaining in right partition to the beginning, without allocating
+                _RANGES _Rotate_one_right(_STD move(_First), _STD move(_Mid), _STD move(++_Last));
+                return;
+            }
+
+            // Find last element in [_Mid, _Last) that is less than *--_Mid
+            const auto _Highest = _Prev_iter(_Mid);
+            while (!_STD invoke(_Pred, _STD invoke(_Proj, *_Last), _STD invoke(_Proj, *_Highest))) {
+                if (_Mid == --_Last) {
+                    // rotate only element remaining in right partition to the beginning, without allocating
+                    _RANGES _Rotate_one_right(_STD move(_First), _STD move(_Mid), _STD move(++_Last));
+                    return;
+                }
+            }
+            ++_Last;
+
+            const iter_difference_t<_It> _Count1 = _RANGES distance(_First, _Mid);
+            if (_Count1 == 1) { // rotate only element remaining in left partition to the end, without allocating
+                _RANGES _Rotate_one_left(_STD move(_First), _STD move(_Mid), _STD move(_Last));
+                return;
+            }
+
+            const iter_difference_t<_It> _Count2 = _RANGES distance(_Mid, _Last);
+            _Optimistic_temporary_buffer<iter_value_t<_It>> _Temp_buf{(_STD min)(_Count1, _Count2)};
+            if (_Count1 <= _Count2 && _Count1 <= _Temp_buf._Capacity) {
+                _RANGES _Inplace_merge_buffer_left(_STD move(_First), _STD move(_Mid), _STD move(_Last),
+                    _Temp_buf._Data, _Temp_buf._Capacity, _STD move(_Pred), _STD move(_Proj));

```

Comment:
We don't move predicates and projections once they've been through `_Pass_fn`, we just copy. (Also on 7933 and 7936; I suggest a search and replace to catch any others that I may have missed.